packages/core/src/decorators/oas2/__tests__/remove-unused-components.test.ts

```diff
@@ -152,4 +152,77 @@ describe('oas2 remove-unused-components', () => {
       },
     });
   });
+
+  it('should remove transitively unused components', async () => {
+    const document = parseYamlToDocument(
+      outdent`
+        swagger: '2.0'
+        paths:
+          /pets:
+            get:
+              produces:
+                - application/json
+              parameters: []
+              responses:
+                '200':
+                  schema:
+                    $ref: '#/definitions/Used'
+              operationId: listPets
+              summary: List all pets
+        definitions:
+          Unused:
+            enum:
+              - 1
+              - 2
```

Comment:
Could you also clean up the tests from the needless fields like that one?

Reply:
This was copied from other test cases, and I am hesitant to include a larger cleanup here.